File: Template/VBSMasterDetail/Component.vb
Public Class Component
    Inherits Srvtools.DataModule

    Friend WithEvents ucMaster As Srvtools.UpdateComponent
    Friend WithEvents ServiceManager As Srvtools.ServiceManager
    Private components As System.ComponentModel.IContainer
    Friend WithEvents Detail As Srvtools.InfoCommand
    Friend WithEvents InfoDataSource1 As Srvtools.InfoDataSource
    Friend WithEvents ucDetail As Srvtools.UpdateComponent
    Friend WithEvents Master As Srvtools.InfoCommand
    Public Sub New(ByVal container As System.ComponentModel.IContainer)
        container.Add(Me)
        InitializeComponent()
    End Sub 'New
    Public Sub New()
        InitializeComponent()
    End Sub 'New
    Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container
        Me.InfoConnection = New Srvtools.InfoConnection
        Me.Master = New Srvtools.InfoCommand(Me.components)
        Me.ServiceManager = New Srvtools.ServiceManager(Me.components)
        Me.ucMaster = New Srvtools.UpdateComponent(Me.components)
        Me.Detail = New Srvtools.InfoCommand(Me.components)
        Me.InfoDataSource1 = New Srvtools.InfoDataSource(Me.components)
        Me.ucDetail = New Srvtools.UpdateComponent(Me.components)
        CType(Me.InfoConnection, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.Master, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.Detail, System.ComponentModel.ISupportInitialize).BeginInit()
        '
        'InfoConnection
        '
        Me.InfoConnection.ConnectionString = Nothing
        Me.InfoConnection.ConnectionType = Srvtools.ConnectionType.SqlClient
        '
        'Master
        '
        Me.Master.CommandText = ""
        Me.Master.CommandTimeout = 30
        Me.Master.CommandType = System.Data.CommandType.Text
        Me.Master.EEPAlias = Nothing
        Me.Master.InfoConnection = Me.InfoConnection
        Me.Master.Name = "Master"
        Me.Master.NotificationAutoEnlist = False
        Me.Master.SecExcept = Nothing
        Me.Master.SecFieldName = Nothing
        Me.Master.SecStyle = Srvtools.SecurityStyle.ssByNone
        Me.Master.SelectTop = 0
        Me.Master.SiteControl = False
        Me.Master.SiteFieldName = Nothing
        Me.Master.Transaction = Nothing
        Me.Master.UpdatedRowSource = System.Data.UpdateRowSource.Both
        '
        'ucMaster
        '
        Me.ucMaster.AutoTrans = True
        Me.ucMaster.ExceptJoin = False
        Me.ucMaster.LogInfo = Nothing
        Me.ucMaster.Name = "ucMaster"
        Me.ucMaster.SelectCmd = Me.Master
        Me.ucMaster.ServerModify = True
        Me.ucMaster.ServerModifyGetMax = False
        Me.ucMaster.TransIsolationLevel = System.Data.IsolationLevel.ReadCommitted
        Me.ucMaster.WhereMode = Srvtools.WhereModeType.Keyfields
        '
        'Detail
        '
        Me.Detail.CommandText = ""
        Me.Detail.CommandTimeout = 30
        Me.Detail.CommandType = System.Data.CommandType.Text
        Me.Detail.EEPAlias = Nothing
        Me.Detail.InfoConnection = Me.InfoConnection
        Me.Detail.Name = "Detail"
        Me.Detail.NotificationAutoEnlist = False
        Me.Detail.SecExcept = Nothing
        Me.Detail.SecFieldName = Nothing
        Me.Detail.SecStyle = Srvtools.SecurityStyle.ssByNone
        Me.Detail.SelectTop = 0
        Me.Detail.SiteControl = False
        Me.Detail.SiteFieldName = Nothing
        Me.Detail.Transaction = Nothing
        Me.Detail.UpdatedRowSource = System.Data.UpdateRowSource.Both
        '
        'InfoDataSource1
        '
        Me.InfoDataSource1.Detail = Me.Detail
        Me.InfoDataSource1.Master = Me.Master
        '
        'ucDetail
        '
        Me.ucDetail.AutoTrans = True
        Me.ucDetail.ExceptJoin = False
        Me.ucDetail.LogInfo = Nothing
        Me.ucDetail.Name = Nothing
        Me.ucDetail.SelectCmd = Me.Detail
        Me.ucDetail.ServerModify = True
        Me.ucDetail.ServerModifyGetMax = False
        Me.ucDetail.TransIsolationLevel = System.Data.IsolationLevel.ReadCommitted
        Me.ucDetail.WhereMode = Srvtools.WhereModeType.Keyfields
        CType(Me.InfoConnection, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.Master, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.Detail, System.ComponentModel.ISupportInitialize).EndInit()

    End Sub
    Friend WithEvents InfoConnection As Srvtools.InfoConnection
End Class

File: AddIns/Templates/SMasterDetail/Component.cs
using System;
using System.ComponentModel;
using System.Collections;
using System.Diagnostics;
using System.Data;
using System.Data.SqlClient;
using System.Xml;
using System.Reflection;
using Microsoft.Win32;
using System.IO;
using Srvtools;
using System.Security;
using System.Security.Permissions;

namespace SMasterDetail
{
    /// <summary>
    /// Summary description for Component.
    /// </summary>
    public class Component : DataModule
    {
        private UpdateComponent ucMaster;
        private ServiceManager serviceManager;
        private InfoCommand Master;
        private InfoCommand Detail;
        private InfoDataSource idsRelation;
        private UpdateComponent ucDetail;
        private InfoConnection infoConnection;
        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.IContainer components;

        public Component(System.ComponentModel.IContainer container)
        {
            ///
            /// Required for Windows.Forms Class Composition Designer support
            ///
            container.Add(this);
            InitializeComponent();

            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }

        public Component()
        {
            ///
            /// This call is required by the Windows.Forms Designer.
            ///
            InitializeComponent();

            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }

        /// <summary> 
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(disposing);
        }

        #region Component Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.components = new System.ComponentModel.Container();
            this.ucMaster = new Srvtools.UpdateComponent(this.components);
            this.Master = new Srvtools.InfoCommand(this.components);
            this.infoConnection = new Srvtools.InfoConnection();
            this.serviceManager = new Srvtools.ServiceManager(this.components);
            this.Detail = new Srvtools.InfoCommand(this.components);
            this.idsRelation = new Srvtools.InfoDataSource(this.components);
            this.ucDetail = new Srvtools.UpdateComponent(this.components);
            ((System.ComponentModel.ISupportInitialize)(this.Master)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.Detail)).BeginInit();
            // 
            // ucMaster
            // 
            this.ucMaster.AutoTrans = true;
            this.ucMaster.ExceptJoin = false;
            this.ucMaster.LogInfo = null;
            this.ucMaster.Name = "ucMaster";
            this.ucMaster.SelectCmd = this.Master;
            this.ucMaster.ServerModify = true;
            this.ucMaster.ServerModifyGetMax = false;
            this.ucMaster.TransIsolationLevel = System.Data.IsolationLevel.ReadCommitted;
            this.ucMaster.WhereMode = Srvtools.WhereModeType.Keyfields;
            // 
            // Master
            // 
            this.Master.CommandText = "";
            this.Master.CommandTimeout = 30;
            this.Master.CommandType = System.Data.CommandType.Text;
            this.Master.EEPAlias = null;
            this.Master.InfoConnection = this.infoConnection;
            this.Master.Name = "Master";
            this.Master.NotificationAutoEnlist = false;
            this.Master.SecExcept = null;
            this.Master.SecFieldName = "";
            this.Master.SecStyle = Srvtools.SecurityStyle.ssByNone;
            this.Master.SelectTop = 0;
            this.Master.SiteControl = false;
            this.Master.SiteFieldName = "";
            this.Master.Transaction = null;
            this.Master.UpdatedRowSource = System.Data.UpdateRowSource.Both;
            // 
            // infoConnection
            // 
            this.infoConnection.ConnectionType = Srvtools.ConnectionType.SqlClient;
            // 
            // Detail
            // 
            this.Detail.CommandText = "";
            this.Detail.CommandTimeout = 30;
            this.Detail.CommandType = System.Data.CommandType.Text;
            this.Detail.EEPAlias = null;
            this.Detail.InfoConnection = this.infoConnection;
            this.Detail.Name = "Detail";
            this.Detail.NotificationAutoEnlist = false;
            this.Detail.SecExcept = null;
            this.Detail.SecFieldName = "";
            this.Detail.SecStyle = Srvtools.SecurityStyle.ssByNone;
            this.Detail.SelectTop = 0;
            this.Detail.SiteControl = false;
            this.Detail.SiteFieldName = "";
            this.Detail.Transaction = null;
            this.Detail.UpdatedRowSource = System.Data.UpdateRowSource.Both;
            // 
            // idsRelation
            // 
            this.idsRelation.Detail = this.Detail;
            this.idsRelation.Master = this.Master;
            // 
            // ucDetail
            // 
            this.ucDetail.AutoTrans = true;
            this.ucDetail.ExceptJoin = false;
            this.ucDetail.LogInfo = null;
            this.ucDetail.Name = "ucDetail";
            this.ucDetail.SelectCmd = this.Detail;
            this.ucDetail.ServerModify = true;
            this.ucDetail.ServerModifyGetMax = false;
            this.ucDetail.TransIsolationLevel = System.Data.IsolationLevel.ReadCommitted;
            this.ucDetail.WhereMode = Srvtools.WhereModeType.Keyfields;
            ((System.ComponentModel.ISupportInitialize)(this.Master)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.Detail)).EndInit();

        }
        #endregion
    }
}
